src/common/endianutil.cpp

```diff
@@ -17,77 +17,21 @@
 
 #include "endianutil.h"
 
-#include <cstdint>
-#include <cstring>
-
-using namespace std;
-
 namespace reone {
 
-template <>
-void swapBytes(uint16_t &val) {
-    val = ((val >> 8) & 0xff) | ((val & 0xff) << 8);
-}
-
-template <>
-void swapBytes(uint32_t &val) {
-    val = ((val >> 24) & 0xff) |
-        (((val >> 16) & 0xff) << 8) |
-        (((val >> 8) & 0xff) << 16) |
-        ((val & 0xff) << 24);
-}
-
-template <>
-void swapBytes(uint64_t &val) {
-    val = ((val >> 56) & 0xff) |
-        (((val >> 48) & 0xff) << 8) |
-        (((val >> 40) & 0xff) << 16) |
-        (((val >> 32) & 0xff) << 24) |
-        (((val >> 24) & 0xff) << 32) |
-        (((val >> 16) & 0xff) << 40) |
-        (((val >> 8) & 0xff) << 48) |
-        ((val & 0xff) << 56);
+Endianess initEndianess() {
+    uint16_t test = 0x1100;
+    return static_cast<bool>(*(char*)&test) ? Endianess::Big : Endianess::Little;
 }
 
-template <>
-void swapBytes(int16_t &val) {
-    val = ((val >> 8) & 0xff) | ((val & 0xff) << 8);
-}
-
-template <>
-void swapBytes(int32_t &val) {
-    val = ((val >> 24) & 0xff) |
-        (((val >> 16) & 0xff) << 8) |
-        (((val >> 8) & 0xff) << 16) |
-        ((val & 0xff) << 24);
-}
-
-template <>
-void swapBytes(int64_t &val) {
-    val = ((val >> 56) & 0xff) |
-        (((val >> 48) & 0xff) << 8) |
-        (((val >> 40) & 0xff) << 16) |
-        (((val >> 32) & 0xff) << 24) |
-        (((val >> 24) & 0xff) << 32) |
-        (((val >> 16) & 0xff) << 40) |
-        (((val >> 8) & 0xff) << 48) |
-        ((val & 0xff) << 56);
-}
+Endianess _end = initEndianess();
```

Comment:
`Endianess g_end = initEndianess()`, this is a global variable. Might be a good idea to make it static too (to put it in file, not global scope).

Reply:
What about static inside function:

```c++
Endianess getEndianess() {
    static Endianess _end = initEndianess();
    return _end;
}
```

Reply:
Fine by me, but then it's just `end`. Or `endianess` to remove ambiguity.